Fix issues, tidy code

Jacky720 opened the pull request:
User changes:
* Resolve #11: Boxes added in new row between Snowdin values, Hotland values, and fun values.
* Resolve #34: All obtainable keys (except the Mystery Key of course) added between Cellphone and Other.
* Resolve #74: Option to download file8 (necessary for post-Photoshop Flowey) added when trapped in Flowey's World. Not currently documented in `help.html`.

Code changes:
* Use `iniIDs` array for writing ini values to JS object (not currently used for printing to the interface).
* Use `flagFor` and `flags` arrays to automatically populate `stateChoiceArrays`, saving redundant information such as the Toriel and Lesser Dog states.
* Use `flagFor` array for printing save values to the interface rather than repeatedly calling `updateSelection`.

Jacky720 committed:
some ini function cleanup

Jacky720 committed:
Add local function to simplify updateIniFromForm

Jacky720 committed:
Add boxes, keys, file8

Not functional yet. Working on #11, #34, and #74.

Jacky720 committed:
[m] Hacky padding fix

Jacky720 committed:
solve three issues and rely more heavily on arrays

Engineer gaming

Jacky720 committed:
Use array + loop instead of function for ini read

Jacky720 committed:
fix room/item filtering

turns out, null does not === undefined

Jacky720 committed:
Remove flags added to basic menu

KockaAdmiralac commented on main.js:
```diff
@@ -847,50 +846,26 @@ function updatePersistentDataForm(iniobj) {
-        if (!ini.reset) {
-            ini.reset = {};
+    
+    for (var i in iniIDs) { // 0 -> element ID, 1 -> ini section, 2 -> ini key
```
Use `forEach` or `for ... of` to iterate over an array:
```javascript
for (const [elementID, iniSection, iniKey] of iniIDs) {
    // ...
}
```
(now that we're IE-incompatible, anyways)

Jacky720 committed:
var -> let and const

And use 'this', and define Blobs inline, and document uneditable values (of which there are like four). Oh, and use ternaries with those checkbox values so that we don't get random Number values in the middle of the save file.

Jacky720 committed:
correct offset in exporting flags

whoops

Jacky720 committed:
[m] keep save file type

whoops x(like 5 or something that's a code moment)

Jacky720 committed:
rm nested functions; working on loadPreset failure

Jacky720 committed:
rv over-zealous "this"ing

one major problem, down. Gonna keep testing.

Jacky720 committed:
global ini + saveLines to fix saving presets

Jacky720 commented:
I'm going to switch around some branches to update https://jacky720.github.io/FloweysTimeMachine/. And close this PR for now.
